### ISSUE TITLE
Date parameters in rules change their value each time you edit and save

### ISSUE THREAD

**Issue by bugfolder** [backdrop-contrib/rules#218]
If you create a Rules plugin that contains a `date` parameter, then the parameter's field says that it's a time in GMT. However,

* The stored date (a timestamp) isn't the GMT date you entered (and it depends on the user time zone setting)
* If you repeatedly edit and save the plugin without changing it, the value of the date changes with each iteration of edit/save.

This can be easily demonstrated using the attached test module, which creates an action, "Date demo" that has a date parameter and reports it to the screen when invoked. 

Install this module, then create a reaction rule that you can trigger (e.g., on event "After updating existing content") and the action "Date demo". Note from the description that the time is GMT.

![image](https://github.com/backdrop-contrib/rules/assets/29214084/7ccf5b38-f1c0-40c5-865d-ab80b44146ca)

Now save the action. Note that it shows the value that was entered:

![image](https://github.com/backdrop-contrib/rules/assets/29214084/dd3fbe71-4e72-4ff9-b770-532ca50b94f2)

Now trigger the action, e.g., by editing and saving a node. The timestamp is displayed:

![image](https://github.com/backdrop-contrib/rules/assets/29214084/7a7651f0-b5bf-42e0-bf51-0f2ee8418d05)

But that timestamp is not 2024-04-01 00:00:00 GMT; it is 00:07:00 GMT, which is 00:00:00 in my time zone, America/Los Angeles (GMT-0700). 

And if I change my time zone to America/New York (GMT-0400) and do the same exercise, the timestamp will be 00:00:00 in that time zone and 00:04:00 in GMT.

What's more, if we go back to the Rules action and click the Edit button, we now see the value set to 2024-04-01 07:00:00, a different value than what we originally set:

![image](https://github.com/backdrop-contrib/rules/assets/29214084/fcda1917-96ad-4de5-b2ab-4d530cd85f14)

And if we save without changing it and then click edit again, it will be different yet again:

![image](https://github.com/backdrop-contrib/rules/assets/29214084/9c7e4246-efab-4d7e-8289-b69ebde35b67)

Each time the time increment is the difference between the user's time zone and GMT.

Interestingly, this behavior is exactly the same in the latest D7 version of Rules.

So it seems like there's at least two bugs here. First, the value that's stored needs to match what the description says it is (so fix either the storing function or the description). Second, the retrieved value needs to be what was originally entered.

[rules_date_demo.zip](https://github.com/backdrop-contrib/rules/files/14946985/rules_date_demo.zip)

**Comment by bugfolder** [comment]
PR submitted. Since the description emphasizes that the time is GMT, that's what users are expecting (even if they haven't been getting it), so I've made it so that the value stored in the GMT timestamp of the time string. And, just to emphasize that, I've made the display of the parameter value on the rules edit page display the time in GMT (rather than in the user's own time zone without time zone indicator, as it was doing before).

![image](https://github.com/backdrop-contrib/rules/assets/29214084/549778b0-583e-471a-93bd-c91e94285d64)

**Comment by argiepiano** [comment]
Good catch. I've tested and seems to be doing the job. 

In the meantime I found an unrelated  inconsistency with date field tokens (since Rules uses tokens to display field values, for example, using "Show message in the site"). I'e posed a question on [Zulip](https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/Date.20fields.20stored.20as.20UTC.20in.20database/near/432805921) about this inconsistency.
